### ISSUE TITLE
Feature request: Ajv formats addFormat support

### ISSUE THREAD

**Issue by cfuerst** [GrantBirki/json-yaml-validate#60]
Hi There 👋 first of all thank you for this great GitHub action ❤️ 

I need to do exactly what you supporting here with the addition that I have custom formats in my schema. 

For example:
```
// in the validator itself:
ajv.addFormat("cron_expression", /(((\d+,)+\d+|(\d+(\/|-)\d+)|\d+|\*) ?){5,7}/)

// json
{
  "some_property_name": { "type": "string", "format": "cron_expression" }
}
```

Could (and would you be open to) you imagine a smart and generic way how this action could support custom formats?
If so I would be happy to try to drive a PR regarding that.

Cheers

**Comment by GrantBirki** [comment]
@cfuerst I would absolutely support this! Please do open a PR and I'll take a look and try to answer any questions you may have 🎉

**Comment by cfuerst** [comment]
What do you think of going with an additional parameter `ajv_custom_regexp_formats` which is handled same as `files` option with `getMultilineInput()` which would look something like this:

```
      - name: json-yaml-validate
        uses: GrantBirki/json-yaml-validate@vX.X.X
        with:
            ajv_custom_regexp_formats: |
                cron_expression=/(((\d+,)+\d+|(\d+(\/|-)\d+)|\d+|\*) ?){5,7}/
                another_custom_format=/[A-z]+/
```

It's not pretty to pass a key value pair like this though. Maybe you have a better idea.
setting any value in `ajv_custom_regexp_formats` would also implicitly set `use_ajv_formats` to `true`.

Feedback greatly appreciated 🙇

**Comment by cfuerst** [comment]
@GrantBirki I added a PR for this https://github.com/GrantBirki/json-yaml-validate/pull/61

**Comment by GrantBirki** [comment]
@cfuerst thank you so much for this feature and being super speedy ⚡!

I have published a new release [`v2.7.0`](https://github.com/GrantBirki/json-yaml-validate/releases/tag/v2.7.0) which contains your new feature. Please give it a go and let me know if it all works okay for you.

If it does work, I'll update the main release `v2` of this Action to pick up your changes there as well.

Thank you! 🙇

**Comment by cfuerst** [comment]
Awesome thank you ill try it out tomorrow asap and give you feedback.

You had quite a quick merge finger here hehe, there are still 2 tests i have not implemented yet which would cover the non happy paths (not sure if they are even needed) - see the last 2 tests from json validator test. Sorry for that, but i can take care of them too. Whats your opinion how to deal with those cases?

**Comment by GrantBirki** [comment]
I was indeed too speedy! If you want to open another PR to fill in those tests cases, that would be great! I don't have any preference as to how they are done. Even without them, I think its working great and that's the reason for acceptance tests too 😃

**Comment by cfuerst** [comment]
https://github.com/GrantBirki/json-yaml-validate/pull/64, ill shortly try out [v2.7.0](https://github.com/GrantBirki/json-yaml-validate/releases/tag/v2.7.0) in the meanwhile

**Comment by cfuerst** [comment]
👍 [v2.7.0](https://github.com/GrantBirki/json-yaml-validate/releases/tag/v2.7.0) works like a charm in my repo

**Comment by GrantBirki** [comment]
Merged https://github.com/GrantBirki/json-yaml-validate/pull/64 🎉 

New release published [`v2.7.1`](https://github.com/GrantBirki/json-yaml-validate/releases/tag/v2.7.1) and I have updated the main tag of `v2` to point to `v2.7.1` via [this workflow](https://github.com/GrantBirki/json-yaml-validate/actions/runs/8745845175) so we should be all set!

Thank you so much for your great contributions here 🙇 ⭐
